File: src/main/java/com/arnas/klatrebackend/interfaces/services/ClimbingSessionService.java
package com.arnas.klatrebackend.interfaces.services;

import com.arnas.klatrebackend.dataclasses.*;

import java.util.List;

public interface ClimbingSessionService {

    ClimbingSession getActiveSessionByGroup(long groupId, long userId);
    List<ClimbingSession> getPastSessionsByGroup(long groupId, long userId);
    void uploadSession(long userId, ClimbingSessionDTO climbingSession);
    ClimbingSession openSession(long groupId, long placeId, long userId);
    void closeSession(long sessionId, boolean save, long userId);
    RouteAttemptDisplay addRouteAttempt(long activeSessionId, long userId, RouteAttemptDTO routeAttempt);
    void removeRouteAttempt(long attemptId, long userId);
    RouteAttempt getRouteAttemptById(long attemptId);
    List<RouteAttemptDisplay> getRouteAttempts(long sessionId);
    void updateRouteAttempt(long userId, RouteAttempt routeAttempt);
    List<ClimbingSessionDisplay> getPastSessions(long groupId, long userId);

}

File: src/main/kotlin/com/arnas/klatrebackend/features/climbingsessions/ClimbingSessionService.kt
package com.arnas.klatrebackend.features.climbingsessions

interface ClimbingSessionService {
    fun getActiveSessionByGroup(groupId: Long, userId: Long): ClimbingSession?
    fun getPastSessionsByGroup(groupId: Long, userId: Long): List<ClimbingSession>
    fun uploadSession(userId: Long, climbingSession: ClimbingSessionDTO)
    fun openSession(groupId: Long, placeId: Long, userId: Long): ClimbingSession
    fun closeSession(sessionId: Long, save: Boolean, userId: Long)
    fun addRouteAttempt(activeSessionId: Long, userId: Long, routeAttempt: RouteAttemptDTO): RouteAttemptDisplay
    fun removeRouteAttempt(attemptId: Long, sessionId: Long, userId: Long)
    fun getRouteAttemptById(attemptId: Long): RouteAttempt
    fun getRouteAttempts(sessionId: Long): List<RouteAttemptDisplay>
    fun updateRouteAttempt(userId: Long, routeAttempt: RouteAttempt)
    fun getPastSessions(groupId: Long, userId: Long): List<ClimbingSessionDisplay>
}

